packages/flutter/lib/src/material/time_picker.dart

```diff
@@ -1475,21 +1199,352 @@ class _DialState extends State<_Dial> with SingleTickerProviderStateMixin {
+    }
+    return null;
+  }
+
+  int _parseMinute(String value) {
+    if (value == null || value.isEmpty) {
+      return null;
+    }
+
+    final int newMinute = int.tryParse(value);
+    if (newMinute == null) {
+      return null;
+    }
+
+    if (newMinute >= 0 && newMinute < 60) {
+      return newMinute;
+    }
+    return null;
+  }
+
+  void _handleHourSavedSubmitted(String value) {
+    final int newHour = _parseHour(value);
+    if (newHour != null) {
+      _selectedTime = TimeOfDay(hour: newHour, minute: _selectedTime.minute);
+      widget.onChanged(_selectedTime);
+    }
+  }
+
+  void _handleHourChanged(String value) {
+    final int newHour = _parseHour(value);
+    if (newHour != null && value.length == 2) {
+      // If a valid hour is typed, move focus to the minute TextField.
+      FocusScope.of(context).nextFocus();
+    }
+  }
+
+  void _handleMinuteSavedSubmitted(String value) {
+    final int newMinute = _parseMinute(value);
+    if (newMinute != null) {
+      _selectedTime = TimeOfDay(hour: _selectedTime.hour, minute: int.parse(value));
+      widget.onChanged(_selectedTime);
+    }
+  }
+
+  void _handleDayPeriodChanged(TimeOfDay value) {
+    _selectedTime = value;
+    widget.onChanged(_selectedTime);
+  }
+
+  String _validateHour(String value) {
+    final int newHour = _parseHour(value);
+    setState(() {
+      hourHasError = newHour == null;
+    });
+    // This is used as the validator for the [TextFormField].
+    // Returning an empty string allows the field to go into an error state.
+    // Returning null means no error in the validation of the entered text.
+    if (newHour == null) {
+      return '';
+    }
+    return null;
+  }
+
+  String _validateMinute(String value) {
+    final int newMinute = _parseMinute(value);
+    setState(() {
+      minuteHasError = newMinute == null;
+    });
+    // This is used as the validator for the [TextFormField].
+    // Returning an empty string allows the field to go into an error state.
+    // Returning null means no error in the validation of the entered text.
+    if (newMinute == null) {
+      return '';
+    }
+    return null;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasMediaQuery(context));
+    final MediaQueryData media = MediaQuery.of(context);
+    final TimeOfDayFormat timeOfDayFormat = MaterialLocalizations.of(context).timeOfDayFormat(alwaysUse24HourFormat: media.alwaysUse24HourFormat);
+    final bool use24HourDials = hourFormat(of: timeOfDayFormat) != HourFormat.h;
+    final ThemeData theme = Theme.of(context);
+    final TextStyle hourMinuteStyle = TimePickerTheme.of(context).hourMinuteTextStyle ?? theme.textTheme.headline2;
+
+    return Padding(
+      padding: const EdgeInsets.symmetric(horizontal: 24.0, vertical: 16.0),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Text(
+            // TODO(rami-a): localize 'ENTER TIME'
```

Comment:
Same comment as before about TODO possibly needing issues. I am unable to find the rule though.

Reply:
Leaving as-is, the rule only requires username, GH issue being optional. I'm going to be following up with a l10n pass anyways